pydarn/pydmap/datastructures.py

```diff
@@ -0,0 +1,212 @@
+        the record if provided.
+
+        Parameter
+        ---------
+        scalar_list: list
+            A list of namedtuples following the DmapScalar structure
+        array_list: list
+            A list of namedtuples following the DmapArray structure
+        """
+        self.__record = OrderedDict()  # private to avoid setting the record.
+        self.num_scalars = 0  # may be useless attributes?
+        self.num_arrays = 0
+
+        # store scalars first into the
+        self._add_scalar_list(scalar_list)
+
+        # store arrays second into the record
+        self._add_array_list(array_list)
+
+    @property
+    def record(self) -> OrderedDict():
+        """
+        A getter for the dmap record
+
+        Return
+        ------
+        __record : OrderedDict
+            returns the private dmap record attribute.
+        """
+        return self.__record
+
+    # TODO If we find num_scalars and num_arrays unused then we
+    # should look into encapsulating more of this.
+    def _add_scalar_list(self, scalar_list=[]):
+        """
+        Private method to iterate over a DmapScalar list to add each
+        DmapScalar namedtuple into the record using the name as the key.
+
+        Parameter
+        ---------
+        scalar_list: list
+            Is a list of DmapScalar namedtuples.
+        """
+
+        for scalar in scalar_list:
+            self.record[scalar.name] = scalar
+            self.num_scalars += 1
+
+    def _add_array_list(self, array_list=[]):
+        """
+        Private method to iterate over a DmapArray list to add each
+        DmapArray namedtuple into the record using the name as the key.
+
+        Parameter
+        ---------
+        array_list: list
+            Is a list of DmapArray namedtuples.
+        """
+        for array in array_list:
+            self.record[array.name] = array
+            self.num_arrays = 0
+
+    def add_scalar(self, scalar):
+        """
+        Adds a DmapScalar or list of DmapScalars to the record.
+
+        Parameter
+        ---------
+        scalar: DmapScalar/list
+            A DmapScalar or a list of DmapScalars namedtuple that is added
+            into the record.
+
+        See Also
+        --------
+        DmapScalar : named tuple for dmap array variable
+        """
+        if isinstance(scalar, list):
+            self._add_scalar_list(scalar)
+        else:
+            self.record[scalar.name] = scalar
+            self.num_scalars += 1
+
+    def add_array(self, array):
+        """
+        Adds a DmapArray or list of DmapArrays to the record.
+
+        Parameter
+        ---------
+        scalar: DmapArray/list
+            A DmapArray or a list of DmapArrays namedtuple that is added
+            into the record.
+
+        See Also
+        ---------
+        DmapArray : named tuple for dmap array variables
+        """
+        if isinstance(array, list):
```

Comment:
Why not dict or tuples? 

In the docstring, I mention that it needs to be a list. Again bittersweet of python. Very non-pythonic to do type checking and to think all possible collections that I could get would just be extraneous code. I don't see the priority in this. 

Plus this class isn't actually used anymore, I just kept it just in case there was a reason to have it.